### PR TITLE
Add check for milestone

### PR DESCRIPTION
**Pull request by hush-hush**
### What does this PR do?

Check that every PR as a milestone.
This is mandatory in order to script the testing phase of our release

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -121,6 +121,17 @@ jobs:
           command: inv -e lint-teamassignment
           name: run PR check for team assignment labels
 
+  milestone:
+    <<: *job_template
+    steps:
+      - restore_cache: *restore_source
+      - restore_cache: *restore_deps
+      - setup_remote_docker
```
Comment (xvello) [inline .circleci/config.yml]: I don't think you need the `setup_remote_docker` directive, that attaches a disposable VM to execute docker commands.
While looking at the config.yml, seems the other lint steps also have this unnecessary directive

File: tasks/test.py [hunk 2]
```diff
@@ -153,6 +153,30 @@ def lint_teamassignment(ctx):
...
+def lint_milestone(ctx):
+    """
+    Make sure PRs are assigned a milestone
+    """
+    pr_url = os.environ.get("CIRCLE_PULL_REQUEST")
+    if pr_url:
+        import requests
+        pr_id = pr_url.rsplit('/')[-1]
+
+        # first check 'noreno' label
```
Comment (olivielpeau) [inline tasks/test.py]: this comment seems irrelevant here:

```suggestion
```

**Review by olivielpeau — COMMENTED** [review]
Just left a nit 👍 

Not necessarily as part of this PR:
* we should document these extra lint steps in our `CONTRIBUTING` doc, so that contributors know in advance how our CI expects PRs to be created :)
* In general, I'm not sure using the PR milestones in a fully automated way is the "right" way to generate a list of all PRs included in a release, because we simply may not set the correct milestone on a PR (especially if we have a CI test that fails on this, we may very well be tempted to just set any milestone to make the CI pass).
  Have we investigated using the git diff between 2 releases to find these PRs? Github seems to be able to link a commit back to the PR that merged it to master, if that's available through the API I think it'd be a more robust solution.
  Happy to discuss this later though.

File: .circleci/config.yml (hunk 1 above)
Comment (hush-hush) [reply .circleci/config.yml]: :+1:

File: tasks/test.py (hunk 2 above)
Comment (hush-hush) [reply tasks/test.py]: I added a line in the contributing page.

For your second point your're right and that's already what we do. But before creating test card having a list of the needed PR for a release is very useful. Once we no that everything we want if merge, we can run the script and create cards automatically.

**Review by olivielpeau — APPROVED** [review]
